src/lib.rs

```diff
@@ -297,6 +299,60 @@ impl Transaction {
 impl_from_core_type!(Transaction, bitcoin::Transaction);
 impl_from_ffi_type!(Transaction, bitcoin::Transaction);
 
+#[derive(Debug, Clone, PartialEq, Eq, uniffi::Object)]
+pub struct Psbt(bitcoin::Psbt);
+
+#[uniffi::export]
+impl Psbt {
+    #[uniffi::constructor]
+    pub fn from_unsigned_tx(tx: Arc<Transaction>) -> Result<Self, PsbtError> {
+        let psbt = bitcoin::Psbt::from_unsigned_tx(tx.0.clone().into())?;
+        Ok(Psbt(psbt))
+    }
+
+    #[uniffi::constructor]
+    pub fn deserialize(psbt_bytes: &[u8]) -> Result<Self, PsbtError> {
+        let psbt = bitcoin::Psbt::deserialize(psbt_bytes)?;
+        Ok(psbt.into())
+    }
+
+    #[uniffi::constructor]
+    pub fn deserialize_base64(psbt_base64: String) -> Result<Self, PsbtParseError> {
+        let psbt = bitcoin::Psbt::from_str(&psbt_base64)?;
+        Ok(psbt.into())
+    }
+
+    pub fn serialize(&self) -> Vec<u8> {
+        self.0.serialize()
+    }
+
+    pub fn serialize_hex(&self) -> String {
+        self.0.serialize_hex()
+    }
+
+    pub fn serialize_base64(&self) -> String {
+        self.0.to_string()
+    }
+
+    pub fn extract_tx(&self) -> Result<Arc<Transaction>, ExtractTxError> {
+        Ok(Arc::new(self.0.clone().extract_tx()?.into()))
+    }
+
+    pub fn combine(&self, other: Arc<Self>) -> Result<Psbt, PsbtError> {
+        let mut psbt = self.0.clone();
+        let other_psbt = other.0.clone();
+        psbt.combine(other_psbt)?;
+        Ok(psbt.into())
+    }
```

Comment:
I opted to not wrap the Psbt in a `Mutex<_>` so this cannot return self, would it be better to wrap the inner psbt in a mutex instead?